main.js

```diff
@@ -0,0 +1,572 @@
+                    heatindexF: weather[1].hourly[0].HeatIndexF,
+                    dewpointF: weather[1].hourly[0].DewPointF,
+                    windchillF: weather[1].hourly[0].WindChillF,
+                    windgustMiles: weather[1].hourly[0].WindGustMiles,
+                    windgustKmph: weather[1].hourly[0].WindGustKmph,
+                },
+                dayAfterTomorrow: {
+                    averageTemp: weather[2].avgtempF,
+                    maxTemp: weather[2].maxtempF,
+                    minTemp: weather[2].mintempF,
+                    totalSnow: weather[2].totalSnow_cm,
+                    precip: weather[2].hourly[0].chanceofrain,
+                    humidity: weather[2].hourly[0].humidity,
+                    wind: weather[2].hourly[0].windspeedKmph,
+                    visibility: weather[2].hourly[0].visibility,
+                    pressure: weather[2].hourly[0].pressure,
+                    cloudcover: weather[2].hourly[0].cloudcover,
+                    uvindex: weather[2].hourly[0].uvIndex,
+                    winddir: weather[2].hourly[0].winddir16Point,
+                    winddirdegree: weather[2].hourly[0].winddirDegree,
+                    windspeed: weather[2].hourly[0].windspeedMiles,
+                    weathercode: weather[2].hourly[0].weatherCode,
+                    weathericon: weather[2].hourly[0].weatherIconUrl[0].value,
+                    weatherdesc: weather[2].hourly[0].weatherDesc[0].value,
+                    precipInches: weather[2].hourly[0].precipInches,
+                    heatindexF: weather[2].hourly[0].HeatIndexF,
+                    dewpointF: weather[2].hourly[0].DewPointF,
+                    windchillF: weather[2].hourly[0].WindChillF,
+                    windgustMiles: weather[2].hourly[0].WindGustMiles,
+                    windgustKmph: weather[2].hourly[0].WindGustKmph,
+                },
+    
+            };
+
+            //console.log('12313212', weatherData.today.weatherDesc);
+            console.log('232112',weather[0].hourly[0].weatherDesc)
+            const weatherDescriptions = weather[0].hourly[0].weatherDesc.map(hour => hour.value.toLowerCase());
+
+            console.log('weatherDescriptions', weatherDescriptions);
+            
+
+
+            //Function to display location or nearest area to the main section
+            function displayLocation() {
+                const userLocation = document.querySelector('#search');
+                LocationName.textContent = userLocation.value;
+                if (weatherData.city.toLowerCase() == userLocation.value.toLowerCase()) {
+                    area.textContent = `Area: \n${weatherData.city}`;
+                } else {
+                    area.textContent = `Nearest Area: \n${weatherData.city}`;
+                }
+                }
+            displayLocation();
+
+            
+            
+            //Display region to the main section
+            function displayRegion() {
```

Comment:
functions like the this one and a few of the ones below can be all one function. There's no added benefit to having multiple functions that execute almost the same thing